js/plates-model/crust.ts

```diff
@@ -1,10 +1,8 @@
 import config from "../config";
 import { random } from "../seedrandom";
-import { BASE_OCEANIC_CRUST_THICKNESS, BASE_CONTINENTAL_CRUST_THICKNESS, FieldType } from "./field";
+import { FieldType } from "./field";
```

Comment:
There still appear to be circular imports between `field.ts` and `crust.ts`.

Reply:
I think there are many circular dependencies of that type in this app (and probably most of our apps). This seems okay to me and I think it's unavoidable in some cases. Quite a common pattern here is to have reference to the parent object in the child object. While it's not perfect, it helps a lot. For example, field properties are greatly based on their parent tectonic plate properties - rotation, angular velocity, etc. In this case, we cannot avoid having `import { Plate } from "./plate"` in `field.ts` and `import { Field } from "./field"` in `plate.ts`.

The problem I was describing was a bit different. We had a set of constants A, B, C and X, Y, Z in two different files. And they're evaluated on the module load like:
```
export const A = X + Y;
export const B = Z;
```
and
```
export const X = A + B;
export const Y = X + A;
```
If it was in one file, TS would complain that some of the constants are used before they're defined. But it went unnoticed as definitions were spread between files. Also, there were some functions involved that made things even more confusing. I've fixed that by moving all the related constants and functions to one file and fixing problems detected by TS (in this commit: https://github.com/concord-consortium/tectonic-explorer/pull/93/commits/c8cace4f706965cc3999de7cf22f2361042b7cbd).